probe-rs/src/config/sequences/atsam.rs

```diff
@@ -331,13 +334,20 @@ impl AtSAM {
     /// # Errors
     /// Subject to probe communication errors
     pub fn release_reset_extension(&self, memory: &mut dyn ArmProbe) -> Result<(), ArmError> {
+        // Enable debug mode if it is not already enabled
+        let mut dhcsr = Dhcsr(0);
+        dhcsr.enable_write();
+        dhcsr.set_c_debugen(true);
+        dhcsr.set_c_halt(true);
```

Comment:
It feels wrong to unconditionally halt the MCU after a reset. What's the reasoning behind this?

Reply:
I don't have better justification than [this](https://github.com/espressif/openocd-esp32/blob/8133bce9617610279a1a1988a6921882610a7409/src/flash/nor/atsame5.c#L885) but the PR seems to work without the halt anyway.

Reply:
Their overall logic might be different :) Not sure. I expect reset to just reset :)